WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,490 @@
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setDisplayShowTitleEnabled(true);
+        }
+        configureSearchView();
+
+        RecyclerView recycler = findViewById(R.id.recycler);
+        recycler.setLayoutManager(new GridLayoutManager(this, getColumnCount()));
+
+        mAdapter = new StockMediaAdapter();
+        recycler.setAdapter(mAdapter);
+
+        mSelectionBar = findViewById(R.id.container_selection_bar);
+        mTextInsert = findViewById(R.id.text_insert);
+        mTextInsert.setOnClickListener(new View.OnClickListener() {
+            @Override public void onClick(View v) {
+                uploadSelection();
+            }
+        });
+        findViewById(R.id.text_clear).setOnClickListener(new View.OnClickListener() {
+            @Override public void onClick(View v) {
+                mAdapter.clearSelection();
+            }
+        });
+
+        setupObserver();
+
+        if (savedInstanceState == null) {
+            mSite = (SiteModel) getIntent().getSerializableExtra(WordPress.SITE);
+            showEmptyView(true);
+        } else {
+            mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
+            mViewModel.readFromBundle(savedInstanceState);
+            if (savedInstanceState.containsKey(KEY_SELECTED_ITEMS)) {
+                ArrayList<Integer> selectedItems = savedInstanceState.getIntegerArrayList(KEY_SELECTED_ITEMS);
+                if (selectedItems != null) {
+                    mAdapter.setSelectedItems(selectedItems);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        if (mSearchView != null) {
+            mSearchView.setOnQueryTextListener(null);
+            mSearchView.setOnCloseListener(null);
+        }
+        super.onDestroy();
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mViewModel.writeToBundle(outState);
+        if (mSite != null) {
+            outState.putSerializable(WordPress.SITE, mSite);
+        }
+        if (!mAdapter.mSelectedItems.isEmpty()) {
+            outState.putIntegerArrayList(KEY_SELECTED_ITEMS, mAdapter.mSelectedItems);
+        }
+    }
+
+    /*
+     * observe when the ViewModel's search results have changed and update the adapter with them
+     */
+    private void setupObserver() {
+        mViewModel.getSearchResults().observe(this, new Observer<List<StockMediaModel>>() {
+            @Override
+            public void onChanged(@Nullable final List<StockMediaModel> mediaList) {
+                if (!isFinishing()) {
+                    if (!mViewModel.isFetching()) {
+                        showProgress(false);
+                    }
+                    //noinspection ConstantConditions
+                    showEmptyView(mediaList.isEmpty()
```

Comment:
If we add the search status as I suggested above ^, I think this should go under that `onChanged` method instead. We don't want to be showing an empty view while a fetch is going on. This will ensure that we don't accidentally show it in some edge cases.